master/reference/cni-plugin/configuration.md

```diff
@@ -180,3 +180,55 @@ When using the CNI `host-local` IPAM plugin, a special value `usePodCidr` is all
 ```
 
 When making use of the `usePodCidr` option, the Calico CNI plugin requires read-only Kubernetes API access to the `Nodes` resource.
+
+##### IPAM Manipulation with Kubernetes Annotations
+
+###### Specifying IP Pools
+
+Calico CNI also supports specifying IP Pools to use to assign the IP address to the pods. 
+Specifying the IP Pools can be done two different ways:
+  1. Specifying `ipv4_pools` and/or `ipv6_pools` in the CNI config file. (See [IPAM](#ipam) section for more details)
+  2. Specifying the IP Pools in the pod manifest through annotations. 
+    * The annotations used to specify the IP Pools are `cni.projectcalico.org/ipv4pools` and `cni.projectcalico.org/ipv6pools` for IPv4 and IPv6 IP Pools respectively.
+
+  Example annotation:
+ 
+```yaml
+annotations:
+      "cni.projectcalico.org/ipv4pools": "[\"192.168.0.0/16\"]"
+```
+
+  > **Note:**
+  >
+  > This requires the IP Pools to exist before `ipv4pools` or `ipv6pools` annotations are used.
+  > Requesting a subset of an IP Pool is not supported. IP Pools requested in the annotations must be the exact match to the IP Pools created.
+
+###### Requesting a Specific IP address
+
+You can also request a specific IP address through Kubernetes annotations with Calico IPAM. 
+There are two annotations to request a specific IP address:
+  1. `cni.projectcalico.org/ipAddrs`: You can request an IPv4 and/or an IPv6 address with this annotation. IPs will be assigned from Calico IPAM, so a backing IP Pool must be created before using this annotation.
```

Comment:
"so a backing IP Pool must" -> "the provided IP addresses must fall within a configured IP Pool"